### ISSUE TITLE
multiple hooks failing

### ISSUE THREAD

**Issue by kirk86** [jwiegley/use-package#797]
I have the following in my config
```
:hook (latex-mode . (turn-on-auto-fill turn-on-reftex))
```
Which produces the error
```
run-hooks(change-major-mode-after-body-hook text-mode-hook TeX-mode-hook LaTeX-mode-hook)
  apply(run-hooks (change-major-mode-after-body-hook text-mode-hook TeX-mode-hook LaTeX-mode-hook))
  run-mode-hooks(text-mode-hook TeX-mode-hook LaTeX-mode-hook)
  TeX-latex-mode()
  apply(TeX-latex-mode nil)
  latex-mode()
  set-auto-mode-0(latex-mode nil)
  set-auto-mode()
  normal-mode(t)
  after-find-file(nil t)
```
It seems that only 
```
:hook ((LaTeX-mode latex-mode) . turn-on-aut-fill)
```
hooks on multiple modes work but not multiple functions or multiple modes to multiple functions.

**Comment by aashwin-rs** [comment]
The syntax that was giving you the error is not supported. It is mentioned under issue [676](https://github.com/jwiegley/use-package/issues/676#issuecomment-387806615) that there are no plans for adding support.

**Comment by kirk86** [comment]
I see, thanks for the pointer.

**Comment by conao3** [comment]
Hi @kirk86, I'm [leaf.el](https://github.com/conao3/leaf.el) author.
FYI, you can use multiple hook using `:mode-hook` keyword.
https://github.com/conao3/leaf-keywords.el#mode-hook-keyword
```emacs-lisp
(leaf latex-mode
  :mode-hook
  (turn-on-auto-fill)
  (turn-on-reftex))
;; macroexpand
;;=> (prog1 'latex-mode
;;     (progn
;;       (defun leaf-keywords-mode-hook--latex-mode--latex-mode-hook nil
;;         "Function autogenerated by leaf-keywords in leaf-block `latex-mode' for hook `latex-mode-hook'."
;;         (turn-on-auto-fill)
;;         (turn-on-reftex))
;;       (add-hook 'latex-mode-hook 'leaf-keywords-mode-hook--latex-mode--latex-mode-hook)))
```

**Comment by kirk86** [comment]
@conao3 Cool, thanks!

**Comment by skangas** [comment]
Closed as duplicate of #676.
